app/facades/food_facade.rb

```diff
@@ -0,0 +1,13 @@
+class FoodFacade
+  def self.search(search_term)
+    response = FoodService.search_foods(search_term)
+    response[:foods].map do |result|
+      FoodPoro.new(result)
+    end
+  end
+
+  def self.total_results(search_term)
+    response = FoodService.search_foods(search_term)
+    response[:totalHits]
+  end
```

Comment:
To follow up on what I mentioned in the controller comment, you could do away with this second method, and have the search method return [response[:totalHits], mapped_results] where mapped_results was the collection of food objects.

Reply:
Thinking back on this now.. I wonder if the fact that you're limiting this in the service call will change the totalHits value. 🤔 If so, then I'd keep with your approach.

Reply:
I left the logic for getting the total results till late in the challenge, so it's a bit slapdash. I think I would want to change this to something like you suggested. the service call returns the whole response from the api, so the totalHits is there as well in the original call.

Reply:
which is to say, have the array be populated in the facade with a further array of foodporos as well as the value of totalHits.